include/rocksdb/filter_policy.h

```diff
@@ -51,22 +51,23 @@ class FilterBitsBuilder {
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
 
   // Calculate num of keys that can be added and generate a filter
-  // <= the specified number of bytes.
-#if defined(_MSC_VER)
-#pragma warning(push)
-#pragma warning(disable : 4702)  // unreachable code
-#endif
-  virtual int CalculateNumEntry(const uint32_t /*bytes*/) {
-#ifndef ROCKSDB_LITE
-    throw std::runtime_error("CalculateNumEntry not Implemented");
-#else
-    abort();
-#endif
-    return 0;
+  // <= the specified number of bytes, though callers (including RocksDB)
+  // should only rely on that for performance, not correctness.
+  // This default implementation is for old API compatibility.
+  virtual size_t CalculateNumEntries(size_t bytes) {
+    bytes = std::min(bytes, size_t{0xffffffff});
+    return static_cast<size_t>(CalculateNumEntry(static_cast<uint32_t>(bytes)));
+  }
+
+  // Old, DEPRECATED version of CalculateNumEntries. This is not
+  // called by RocksDB except as the default implementation of
+  // CalculateNumEntries for API compatibility.
+  virtual int CalculateNumEntry(const uint32_t bytes) {
+    // DEBUG: ideally should not rely on this implementation
+    assert(false);
+    // RELEASE: something reasonably "safe": 2 bytes per entry
+    return static_cast<int>(bytes / 2);
```

Comment:
sorry, why 2 bytes per entry is safe? Isn't it change the default behavior of the API?

Reply:
The old behavior was to throw an exception, which in RocksDB can lead to UB including DB corruption. abort() would be better, but since we don't really expect outside callers of this API, just implementors, I've added this:

    // callers (including RocksDB)
    // should only rely on that for performance, not correctness.

Thus, it's ok for correctness to return something wrong. 16 bits per key is probably not too far from reality. Perhaps I should say "conservative" instead of "`"safe"`"